### PR TITLE
Performance: measure typing without inspector

### PR DESCRIPTION
**Pull request by ellatrix**
## What?

Adds a typing perf metric with the inspector panel closed.

## Why?

* We are trying promote the isolated block editor, which is often used without an inspector panel, so excellent performance would be great here (I believe we can get this metric even further down).
* Catch regressions: for example creating react trees for inspector controls that won't be rendered is bad.
* The difference in metrics gives us a good picture of how much of the performance issues are coming from the inspector.

### CODE REVIEW

File: test/performance/config/performance-reporter.ts [hunk 1]
```diff
@@ -48,6 +49,9 @@ export interface WPPerformanceResults {
 	type?: number;
 	minType?: number;
 	maxType?: number;
+	typeWithoutInspector?: number;
+	minTypeWithoutInspector?: number;
+	maxTypeWithoutInspector?: number;
```
Comment (youknowriad) [inline test/performance/config/performance-reporter.ts]: I'd remove the min/max to be honest, not that useful.

File: test/performance/specs/post-editor.spec.js [hunk 2]
```diff
@@ -166,37 +198,7 @@ test.describe( 'Post Editor Performance', () => {
...
-			const [ keyDownEvents, keyPressEvents, keyUpEvents ] =
-				metrics.getTypingEventDurations();
-
-			// Save the results.
-			for ( let i = throwaway; i < iterations; i++ ) {
-				results.typeContainer.push(
-					keyDownEvents[ i ] + keyPressEvents[ i ] + keyUpEvents[ i ]
-				);
-			}
+			await type( firstParagraph, metrics, 'typeContainer' );
```
Comment (youknowriad) [inline test/performance/specs/post-editor.spec.js:201]: I intentionally kept the way we compute "type in container" different than the regular "type". 

One types and waits, the other doesn't wait. My thinking was that both of these metrics are useful but they are different. This PR reverts that.

Comment (youknowriad) [reply test/performance/specs/post-editor.spec.js:201]: Reading the diff, maybe this has been reverted elsewhere?

Comment (youknowriad) [reply test/performance/specs/post-editor.spec.js:201]: ok never mind, it was reverted before.

File: test/performance/config/performance-reporter.ts (hunk 1 above)
Comment (ellatrix) [reply test/performance/config/performance-reporter.ts]: Sure, can remove it

**Review by youknowriad — APPROVED** [review]

**Comment by youknowriad** [comment]
This had an impact on some numbers like "type in container", I guess this changed a bit how we compute the metric. Do you know what changed exactly?

**Comment by youknowriad** [comment]
Also, why do you think this PR seem to have impacted "block select" metric as well?

**Comment by youknowriad** [comment]
cc @WunderBart might also be interested in this PR

File: test/performance/specs/post-editor.spec.js (hunk 2 above)
Comment (WunderBart) [reply test/performance/specs/post-editor.spec.js:201]: Typing metric computation was made consistent within https://github.com/WordPress/gutenberg/pull/52022. I decided to opt for waiting between keystrokes following your suggestion [here](https://github.com/WordPress/gutenberg/issues/51383#issuecomment-1584893744).

**Comment by WunderBart** [comment]
> This had an impact on some numbers like "type in container", I guess this changed a bit how we compute the metric. Do you know what changed exactly?

> Also, why do you think this PR seem to have impacted "block select" metric as well?

@youknowriad, if [closing the Editor Settings panel](https://github.com/WordPress/gutenberg/pull/56753/files#diff-7ec757461fce833ba548841172950df439a392c3de98d773a64c77b2174ae429R165-R167) is persistent for the current editor instance, it will likely affect all the subsequent tests. It should be reopened after the test to keep the other numbers stable.

_edit: Nevermind, just noticed https://github.com/WordPress/gutenberg/pull/56780_

**Pull request merged**